pipelines/google_ads_ingestion/utils/add_all_updated_ads.py

```diff
@@ -0,0 +1,100 @@
+    records are added based on raw data changes. This maintains historical versions of each ad.
+
+    Args:
+        bigquery_client (bigquery.Client): An instance of BigQuery client to execute queries.
+        project_id (str): The Google Cloud project ID where BigQuery datasets reside.
+        dataset_id (str): The ID of the dataset containing both the target and tracking tables.
+        raw_table_id (str): The ID of the raw table where updated ad data is stored.
+
+    Returns:
+        IngestionStatus: An enum value indicating the status of the insertion process:
+            - DATA_INSERTED: New rows were successfully added to the raw table.
+            - NO_NEW_UPDATES: No new rows were added, as all ads already existed in the table.
+
+    """
+    initial_row_count = check_table_row_count(
+        bigquery_client, project_id, dataset_id, raw_table_id
+    )
+
+    query = f"""
+    INSERT INTO `{project_id}.{dataset_id}.{raw_table_id}` 
+    (data_modified, metadata_time, advertiser_id, creative_id, raw_data)
+
+    WITH filtered_ads AS (
+        SELECT 
+            TIMESTAMP(PARSE_DATE('%Y-%m-%d', (SELECT region.first_shown FROM UNNEST(t.region_stats) AS region WHERE region.region_code = "SE"))) AS data_modified,
+            CURRENT_TIMESTAMP() AS metadata_time,
+            t.advertiser_id,
+            t.creative_id,
+            TO_JSON_STRING(STRUCT(
+                t.advertiser_id,
+                t.creative_id,
+                t.creative_page_url,
+                t.ad_format_type,
+                t.advertiser_disclosed_name,
+                t.advertiser_legal_name,
+                t.advertiser_location,
+                t.advertiser_verification_status,
+                t.topic,
+                t.is_funded_by_google_ad_grants,
+                ARRAY(
+                    SELECT AS STRUCT *
+                    FROM UNNEST(t.region_stats) AS region
+                    WHERE region.region_code = "SE"
+                ) AS region_stats,
+                t.audience_selection_approach_info
+            )) AS raw_data
+        FROM
+            `bigquery-public-data.google_ads_transparency_center.creative_stats` AS t
+        WHERE 
+            t.advertiser_id IN (SELECT advertiser_id FROM `{project_id}.{dataset_id}.{ADVERTISERS_TRACKING_TABLE_ID}`)
+            AND t.advertiser_location = "SE"
+            AND EXISTS (
+                SELECT 1 FROM UNNEST(t.region_stats) AS region WHERE region.region_code = "SE"
+            )
+    )
+    SELECT 
+        data_modified,
+        metadata_time,
+        advertiser_id,
+        creative_id,
+        raw_data
+    FROM
+        filtered_ads
+    WHERE NOT EXISTS (
+        SELECT 1
+        FROM `{project_id}.{dataset_id}.{raw_table_id}` AS existing
+        WHERE existing.advertiser_id = ads_with_dates.advertiser_id
+        AND existing.creative_id = ads_with_dates.creative_id
+        AND existing.raw_data = ads_with_dates.raw_data
+    )
```

Comment:
try not to have sql queries in your python code directly, especially big one like this. and because you use a lot of queries like this what I'ld do is to create file for the queries and create a builder class for them
```python
from .queries import * # a file with your queries templates
class QueryBuilder:
    @staticmethod
    def build_add_updated_ads_class(your, parameters, for, this, query):
        return ADD_UPDATED_ADS.format(your, parameters)
```

I think this wou'ld be good for starters here